src/overlays/actors/ovl_En_Osn/z_en_osn.c

```diff
@@ -74,59 +101,841 @@ static DamageTable D_80AD2550 = {
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    this->unk_1FC = -1;
+    switch (this->actor.params & 3) {
+        case 0:
+            if (((gSaveContext.entranceIndex == 0xC020) || (gSaveContext.entranceIndex == 0xC030)) ||
+                (gSaveContext.entranceIndex == 0xC060)) {
+                this->unk_1EA = (u16)(this->unk_1EA | 1);
+            }
+            this->unk_1F0 = 1;
+            if (globalCtx->sceneNum == 0x63) {
+                if ((gSaveContext.entranceIndex == 0xC020) || (gSaveContext.entranceIndex == 0xC060)) {
+                    this->actionFunc = func_80AD16A8;
+                    return;
+                }
+                if (gSaveContext.entranceIndex == 0xC030) {
+                    func_80AD1398(this);
+                    this->actionFunc = func_80AD1634;
+                    return;
+                }
+                func_80AD144C(this, globalCtx);
+                return;
+            }
+            func_80AD144C(this, globalCtx);
+            return;
+
+        case 1:
+            this->unk_1EC = 0xF;
+            Actor_ChangeAnimation(&this->skelAnime, sAnimations, this->unk_1EC);
+            this->actionFunc = EnOsn_Idle;
+            return;
+
+        case 2:
+            this->unk_1EC = 0x10;
+            Actor_ChangeAnimation(&this->skelAnime, sAnimations, this->unk_1EC);
+            this->actionFunc = EnOsn_Idle;
+            return;
+
+        case 3:
+            this->actor.flags &= ~ACTOR_FLAG_1;
+            this->actionFunc = func_80AD16A8;
+            return;
+
+        default:
+            Actor_MarkForDeath(&this->actor);
+            return;
+    }
+}
+
+void EnOsn_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnOsn* this = THIS;
+
+    SkelAnime_Free(&this->skelAnime, globalCtx);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnOsn_Update(Actor* thisx, GlobalContext* globalCtx) {
+    u32 pad1;
+    u16 pad2;
+    u32 sp34;
+    EnOsn* this = THIS;
+
+    sp34 = Flags_GetSwitch(globalCtx, 0);
+    this->actionFunc(this, globalCtx);
+    Actor_MoveWithGravity(&this->actor);
+    SkelAnime_Update(&this->skelAnime);
+    if ((this->actor.params & 3) == 0) {
+        if (sp34 != 0) {
+            this->actor.flags |= ACTOR_FLAG_1;
+            func_80AD0830(this, globalCtx);
+            this->actor.draw = EnOsn_Draw;
+        } else {
+            this->actor.draw = NULL;
+            this->actor.flags &= ~ACTOR_FLAG_1;
+        }
+    }
+    func_800E9250(globalCtx, &this->actor, &this->unk_1D8, &this->unk_1DE, this->actor.focus.pos);
+}
+
+s32 EnOsn_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* actor,
+                           Gfx** gfx) {
+    EnOsn* this = (EnOsn*)actor;
+
+    if (this->unk_1F0 && limbIndex == 0xB) {
```

Comment:
```suggestion
    if (this->unk_1F0 && limbIndex == 11) {
```
limbIndex can be in decimal